src/array/attribute.rs

```diff
@@ -383,4 +445,33 @@ mod test {
             .expect("Error getting cell val size.");
         assert_eq!(size, u64::MAX);
     }
+
+    #[test]
+    fn attribute_test_set_fill_value_error() -> TileDBResult<()> {
+        let ctx = Context::new()?;
+        let attr = Attribute::new(&ctx, "foo", Datatype::UInt32)?;
+        attr.set_nullable(&ctx, true)?;
+
+        assert!(attr.set_fill_value(&ctx, 5_i32).is_err());
+        assert!(attr.set_fill_value(&ctx, 5_u32).is_err());
+        assert!(attr.set_fill_value(&ctx, 1.0_f64).is_err());
+
+        Ok(())
+    }
+
+    #[test]
+    fn attribute_test_set_fill_value() -> TileDBResult<()> {
```

Comment:
We will definitely want string tests soon but I think we can skip for now - I left a related comment in `convert.rs`.